src/functions.jl

```diff
@@ -86,11 +99,47 @@ The scalar-valued affine function ``a^T x + b``, where:
 Duplicate variable indices in `terms` are accepted, and the corresponding
 coefficients are summed together.
 """
-mutable struct ScalarAffineFunction{T} <: AbstractScalarFunction
+mutable struct ScalarAffineFunction{T} <: AbstractScalarAffineFunction{T}
     terms::Vector{ScalarAffineTerm{T}}
     constant::T
 end
 
+"""
+    ScalarAffineColumnFunction{T}
+
+Represents a scalar-valued affine function ``a^T x + b``.
+Unlike `ScalarAffineFunction`, it maintains the coefficients and variables in separate vectors.
+"""
+mutable struct ScalarAffineColumnFunction{T} <: AbstractScalarAffineFunction{T}
```

Comment:
Why calling it `column` ?

Reply:
Excellent question, I couldn't find a non-terrible name so I just picked one arbitrary for now. Actually this could be more `ScalarAffineRowFunction` because you represent the row of coefficients